### ISSUE TITLE
escapeMarkdown does not escape spoiler syntax in messages

### ISSUE THREAD

**Issue by sauerbraten** [discordjs/discord.js#3048]
**Please describe the problem you are having in as much detail as possible:**

The newly introduced spoiler tag syntax is not yet handled in Util.escapeMarkdown. For example, `||bla||` would show up exactly like that before and now appears as a spoiler around `bla`, even after running it through escapeMarkdown().

**Include a reproducible code sample here, if possible:**

```js
console.log(Discord.escapeMarkdown('||test||'))

// expected: \|\|test\|\| or \||test\|| (both work)
// actually printed: ||test||
```

**Further details:**
- discord.js version: 11.4.2

Not tested on latest master, but the code looks like it doesn't handle pipes at all: https://github.com/discordjs/discord.js/blob/42505b78c15306491ac3a605f86a3e010bc80958/src/util/Util.js#L84

**Comment by appellation** [comment]
will be fixed in #3042

**Comment by sauerbraten** [comment]
Sweet thanks! Well, now there is an issue for it as well...

**Comment by BannerBomb** [comment]
Can't wait for spoilers to be added to it because I use escapemarkdown and I don't feel like going through and applying a fix to it myself to each place the bot does something with messages.

**Comment by Darqam** [comment]
If it's that urgent you could always make the changes shown in the PR locally.

At the moment there's a bit more talk around it so it's not merged right away, but nothing stops you from editing your local version with that in the mean time.

**Comment by kyranet** [comment]
Sorry for the delay, this landed in master with #3241 in 00c4098bb315a92322d708fe4cf71d070508ab69 😄
